ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/eth1/Eth1Address.java

```diff
@@ -27,17 +32,93 @@ public class Eth1Address extends Bytes20 {
           .format("byte")
           .build();
 
-  public static final Eth1Address ZERO = new Eth1Address(Bytes.wrap(new byte[SIZE]));
+  public static final Eth1Address ZERO =
+      new Eth1Address("0x0000000000000000000000000000000000000000");
+
+  private final String address;
+
+  private Eth1Address(String value) {
+    super(Bytes.fromHexString(value));
+    this.address = value;
+  }
 
-  public Eth1Address(final Bytes bytes) {
-    super(bytes);
+  private static void checkAddress(String value) {
+    checkArgument(value.startsWith("0x"), "Eth1Address (%s) should start with \"0x\".", value);
+    Bytes bytes = Bytes.fromHexString(value);
+    checkArgument(
+        bytes.size() == SIZE, "Eth1Address should be 20 bytes, but was %s bytes.", bytes.size());
   }
 
   public static Eth1Address fromHexString(String value) {
-    return new Eth1Address(Bytes.fromHexString(value));
+    checkAddress(value);
+    Eth1Address address = new Eth1Address(value);
+    Eth1Address encodedAddress = fromHexStringWithChecksum(value);
+    checkArgument(
+        address.equals(encodedAddress),
+        "Eth1Address fails checksum:\n got: %s\n exp: %s",
+        address,
+        encodedAddress);
```

Comment:
oh I see, `fromHexStringWithChecksum` is reformatting to add the checksum. 

So I think there's a few changes we should make here:

1. If the supplied address is either all upper or all lower case we should accept it without checking the checksum. It's very common for users to have addresses that don't have the checksum embedded and quite hard for them to calculate the checksum so requiring it is a bit harsh. If there's mixed case we enforce the checksum which I believe matches how MetaMask handles this.
2. When we print addresses we should always add the checksum. So if the user supplies the value all-lower case, when we print it it will come back as mixed case with the checksum correctly computed.
3. We probably want to avoid creating the `Bytes` instance multiple times. So I'd modify the `Eth1Address` constructor to take both `Bytes` and `String`, then we can parse to `Bytes` and check the length is right, calculate and potentially check the hash code here and then pass the pre-built `Bytes` and mixed-case `String` into the constructor.
4. We should then be able to remove `fromHexStringWithChecksum` and probably just have a private method that takes a `String` and returns it formatted with the correct checksummed case.

And I probably wouldn't require the `0x` prefix. I'd be surprised if people don't have that but no real reason to require it.